plan/planbuilder.go

```diff
@@ -453,18 +463,119 @@ func (b *planBuilder) buildSimple(node ast.StmtNode) Plan {
 	return &Simple{Statement: node}
 }
 
+func (b *planBuilder) getDefaultValue(col *table.Column) (*expression.Constant, error) {
+	if value, ok, err := table.GetColDefaultValue(b.ctx, col.ToInfo()); ok {
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		return &expression.Constant{Value: value, RetType: &col.FieldType}, nil
+	}
+	return &expression.Constant{RetType: &col.FieldType}, nil
+}
+
+func (b *planBuilder) findDefaultValue(cols []*table.Column, name *ast.ColumnName) (*expression.Constant, error) {
+	for _, col := range cols {
+		if col.Name.L == name.Name.L {
+			return b.getDefaultValue(col)
+		}
+	}
+	return nil, ErrUnknownColumn.GenByArgs(name.Name.O, "field_list")
+}
+
 func (b *planBuilder) buildInsert(insert *ast.InsertStmt) Plan {
+	// Get Table
+	ts, ok := insert.Table.TableRefs.Left.(*ast.TableSource)
+	if !ok {
+		b.err = errors.New("Can not get table")
+		return nil
+	}
+	tn, ok := ts.Source.(*ast.TableName)
+	if !ok {
+		b.err = errors.New("Can not get table")
+		return nil
+	}
+	tableInfo := tn.TableInfo
+	schema := expression.TableInfo2Schema(tableInfo)
+	table, ok := b.is.TableByID(tableInfo.ID)
+	if !ok {
+		b.err = errors.Errorf("Can't get table %s.", tableInfo.Name.O)
+		return nil
+	}
 	insertPlan := &Insert{
-		Table:           insert.Table,
+		Table:           table,
 		Columns:         insert.Columns,
-		Lists:           insert.Lists,
-		Setlist:         insert.Setlist,
-		OnDuplicate:     insert.OnDuplicate,
+		tableSchema:     schema,
 		IsReplace:       insert.IsReplace,
 		Priority:        insert.Priority,
 		Ignore:          insert.Ignore,
 		baseLogicalPlan: newBaseLogicalPlan(Ins, b.allocator),
 	}
+	cols := table.Cols()
+	for _, valuesItem := range insert.Lists {
+		exprList := make([]expression.Expression, 0, len(valuesItem))
+		for i, valueItem := range valuesItem {
+			var expr expression.Expression
+			var err error
+			if dft, ok := valueItem.(*ast.DefaultExpr); ok {
+				if dft.Name != nil {
+					expr, err = b.findDefaultValue(cols, dft.Name)
+				} else {
+					expr, err = b.getDefaultValue(cols[i])
+				}
+			} else {
+				expr, _, err = b.rewrite(valueItem, nil, nil, true)
```

Comment:
We can optimize `rewrite`, if valueItem is `ast.ValueExpr`, we can convert to constant directly.
Because when importing data, there are a lot of constant expression.